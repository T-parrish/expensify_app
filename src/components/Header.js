import React from 'react';
import {NavLink} from 'react-router-dom'

const Header = () => (
	<div>
		<h1>Expensify!</h1>
		<ol>
			<li><NavLink to='/' activeClassName="is-active" exact={true}>Home page</NavLink></li>
			<li><NavLink to='/create' activeClassName="is-active">Create Expense</NavLink></li>
			<li><NavLink to='/help' activeClassName="is-active">Help Page</NavLink></li>
		</ol>
	</div>
);

export default Header;