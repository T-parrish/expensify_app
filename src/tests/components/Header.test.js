import React from 'react';
// react-test-renderer
// shallow render doesn't render child components
// import ReactShallowRenderer from 'react-test-renderer/shallow';

// Don't need this line after adding snapshot serializer to config
// import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Header from '../../components/Header';


test('should render Header correctly', () => {
    const wrapper = shallow(<Header />);
    // pre enzymeToJson format
    // expect(wrapper.find('h1').text()).toBe('Expensify!');
    // pre serializer in jest config format
    // expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
});

// // syntax for ReactShallowRenderer
// test('should render Header correctly', () => {
//     const renderer = new ReactShallowRenderer();
//     renderer.render(<Header />);
//     expect(renderer.getRenderOutput()).toMatchSnapshot();
//     console.log(renderer.getRenderOutput());
// });