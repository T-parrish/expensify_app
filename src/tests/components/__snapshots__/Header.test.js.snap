// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div>
  <h1>
    Expensify!
  </h1>
  <ol>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Home page
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/create"
      >
        Create Expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/help"
      >
        Help Page
      </NavLink>
    </li>
  </ol>
</div>
`;
